pkg/api/cached.go

```diff
@@ -239,6 +255,17 @@ func (c *Cached) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 		return nil, fmt.Errorf("failed to change permissions of app directory %s: %v", path.Join(targetPath, "app"), err)
 	}
 
+	// For testing where we're not running as root, we need to chown the app dir via the command line :(
+	if appUser != NO_CHANGE_USER && os.Getenv("RUN_WITH_SUDO") == "true" {
+		err = execCommand("chown", "-R", fmt.Sprintf("%d:%d", appUser, appGroup), path.Join(targetPath, "app"))
+	} else {
+		err = os.Chown(path.Join(targetPath, "app"), appUser, appGroup)
+	}
```

Comment:
Are we supposed to `chown` if `appUser == NO_CHANGE_USER`? i.e. should we do this instead?

```suggestion
	if appUser != NO_CHANGE_USER {
		if os.Getenv("RUN_WITH_SUDO") == "true" {
			err = execCommand("chown", "-R", fmt.Sprintf("%d:%d", appUser, appGroup), path.Join(targetPath, "app"))
		} else {
			err = os.Chown(path.Join(targetPath, "app"), appUser, appGroup)
		}
	}
```

Reply:
From the go docs if `appUser == NO_CHANGE_USER` then the go call no-ops, but that's more readable, I'll change it.